src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -294,7 +293,23 @@ std::pair<SQLPipelineStatus, const std::shared_ptr<const Table>&> SQLPipelineSta
 
   // Get output from the last task if the task was an actual operator and not a transaction statement
   if (!_is_transaction_statement()) {
-    const auto& last_executed_operator = static_cast<const OperatorTask&>(*tasks.back()).get_operator();
+    if constexpr (HYRISE_DEBUG) {
+      auto last_executed_operator_count{0};
+      for (const auto& task : tasks) {
+        const auto& current_operator = static_cast<const OperatorTask&>(*task).get_operator();
+        if (current_operator->state() == OperatorState::ExecutedAndAvailable) last_executed_operator_count++;
+      }
+      Assert(last_executed_operator_count == 1,
```

Comment:
You used `find_if` below. Have you met his friend, `count_if`? ;)

You could even re-use the lambda.